src/pages/List.js

```diff
@@ -42,21 +57,26 @@ export default function List({ token }) {
             </section>
           ) : (
             <ul>
-              {listItem.docs.map((doc) => (
-                <li key={doc.id} className="checkbox-wrapper">
-                  <label>
-                    <input
-                      type="checkbox"
-                      defaultChecked={compareTimeStamps(
-                        doc.data().last_purchased,
-                      )}
-                      disabled={compareTimeStamps(doc.data().last_purchased)}
-                      onClick={(e) => markItemPurchased(e, doc.id)}
-                    />
-                    {doc.data().item_name}
-                  </label>
-                </li>
-              ))}
+              {listItem.docs
+                .filter(
+                  (doc) =>
+                    doc.data().item_name.includes(filter) || filter === '',
+                )
+                .map((doc, index) => (
+                  <li key={doc.id} className="checkbox-wrapper">
+                    <label for={`grocery-item${++index}`}>
+                      <input
+                        type="checkbox"
+                        defaultChecked={compareTimeStamps(
+                          doc.data().last_purchased,
+                        )}
+                        disabled={compareTimeStamps(doc.data().last_purchased)}
+                        onClick={(e) => markItemPurchased(e, doc.id)}
+                      />
+                      {doc.data().item_name}
+                    </label>
```

Comment:
We figured it might be good to number the items on our list so each label would have its own individual for element, but it's no big deal, we can remove it.